packages/gatsby-theme-mdx/src/components/edit-link.js

```diff
@@ -8,7 +8,7 @@ export default () => (
   <Location>
     {({location}) => (
       <a
-        href={base + location.pathname + '.mdx'}
+        href={base + location.pathname.replace(/\/$/, '') + '.mdx'}
```

Comment:
Pages with trailing slash are index pages, so this would be the correct fix:

```suggestion
        href={base + location.pathname.replace(/\/$/, '/index') + '.mdx'}
```